Wikipedia/Code/WMFNearbySectionController.m

```diff
@@ -145,14 +158,50 @@ - (NSString*)analyticsContentType {
     return @"Nearby";
 }
 
+- (WMFLocationManager*)currentLocationManager {
+    if (_currentLocationManager == nil) {
+        _currentLocationManager          = [WMFLocationManager fineLocationManager];
+        _currentLocationManager.delegate = self;
+    }
+    return _currentLocationManager;
+}
+
+- (void)nearbyController:(WMFLocationManager*)controller didUpdateLocation:(CLLocation*)location {
+    self.location = location;
+    currentLocationResolver(self.location);
+    [self.currentLocationManager stopMonitoringLocation];
+}
+
+- (void)nearbyController:(WMFLocationManager*)controller didUpdateHeading:(CLHeading*)heading {
+}
+
+- (void)nearbyController:(WMFLocationManager*)controller didReceiveError:(NSError*)error {
+}
+
 - (AnyPromise*)fetchData {
+    AnyPromise *nearbyTitlesPromise = nil;
     if ([self fetchedResultsAreCloseToLocation:self.location]) {
         DDLogVerbose(@"Not fetching nearby titles for %@ since it is too close to previously fetched location: %@.",
                      self.location, self.searchResults.location);
-        return [AnyPromise promiseWithValue:self.items];
+        nearbyTitlesPromise = [AnyPromise promiseWithValue:self.items];
+    }else{
+        nearbyTitlesPromise = [self fetchTitlesForLocation:self.location];
     }
 
-    return [self fetchTitlesForLocation:self.location];
+    if ([self.date isToday]){
+
+        dispatchOnMainQueueAfterDelayInSeconds(0.1, ^{
+            [self.currentLocationManager startMonitoringLocation];
+        });
+        
+        return [[AnyPromise alloc] initWithResolver:&currentLocationResolver].then(^(PMKResolver resolve) {
```

Comment:
@coreyfloyd Wondering if I should just use https://github.com/mxcl/PromiseKit/blob/master/Categories/CoreLocation/CLLocationManager%2BAnyPromise.h instead of this new `currentLocationManager`...